client/app/hearings/utils.js

```diff
@@ -480,7 +480,81 @@ export const formatChangeRequestType = (type) => {
   }
 };
 
-export const dispositionLabel = (disposition) => HEARING_DISPOSITION_TYPE_TO_LABEL_MAP[disposition] ?? 'None'
+export const dispositionLabel = (disposition) => HEARING_DISPOSITION_TYPE_TO_LABEL_MAP[disposition] ?? 'None';
+
+/**
+ * Method to set the available time slots based on the hearings scheduled
+ * @param {array} hearings -- List of hearings scheduled for a specific date
+ */
+export const setTimeSlots = (hearings) => {
+  // Default to using EST for all times before conversion
+  moment.tz.setDefault('America/New_York');
+
+  // Establish the time period and increment to create time slots between (15:00 - 8:00 and add 1 to include 15:00)
+  const hoursInPeriod = 8;
+  const incrementAmount = 30;
+  const startTime = 8;
+
+  // Safe assign the hearings array in case there are no scheduled hearings
+  const scheduledHearings = hearings || [];
+
+  // Store a list of the filled times to compare against when creating the time slots
+  const filledTimes = scheduledHearings.map((hearing) => {
+    const [hour, minute] = hearing.hearingTime.split(':');
+
+    // Return the time value with the hours and minutes for the time
+    return new Date(`01/01/1900 ${hour}:${minute}`);
+  });
+
+  // Loop up to the number of hours in the period to get the available times
+  const availableTimes = _.compact(_.times(hoursInPeriod).map((index) => {
+    // Add the index to the start time so we assign 1 value per hour
+    const hour = startTime + index;
+
+    // Convert the hour and increment amount to a date so we can compare
+    const availableTime = new Date(`01/01/1900 ${hour}:${incrementAmount}`);
+
+    // Get the difference between the time available and the scheduled times
+    const filled = filledTimes.filter((time) => {
+      // Calculate the time difference then divide by the number of seconds and divide by 60 to get minutes
+      const interval = Math.floor((availableTime - time) / 1000 / 60);
+
+      // Return true if the scheduled time is within an hour of the time slot
+      return interval >= 0 && interval <= 60;
```

Comment:
```suggestion
      return interval >= 0 && interval < 60;
```
i think this should be strictly less than 60 because otherwise it does not show the `10:30` slot.
<img width="760" alt="Screen Shot 2021-03-01 at 11 15 59 AM" src="https://user-images.githubusercontent.com/20735998/109525468-8c681a00-7a7f-11eb-9525-6b452e2bda1a.png">